core/services/fluxmonitorv2/integrations_test.go

```diff
@@ -573,6 +573,10 @@ ds1 -> ds1_parse
 		completesAnswer: false,
 	})
 
+	// Logs from log broadcaster are sent only after a next block is received.
+	time.Sleep(2 * time.Second)
```

Comment:
yeah i did not find a way to avoid this yet (nor lower the sleep time) - perhaps i need to await the next head, but not sure - will see more

Reply:
Can we not subscribe to the log broadcaster in the test and wait for the requisite log to come through?

Reply:
the test already does that, but the issue is: the log will not come unless another head is sent, and not too early, not too late:
```
        // The wait time must be sufficient to allow for flux monitor to finish Register process in log broadcaster
	// and then to have log broadcaster backfill logs after the debounceResubscribe period of ~ 1 sec
	time.Sleep(2 * time.Second)
	// Finally, the logs from log broadcaster are sent only after a next block is received.
	fa.backend.Commit()
```

so to avoid the Sleep, we would have to do something like: `app.LogBroadcaster.AwaitNextDebounceAndBackfill()` and block on that - i'll see what can be done.

Reply:
ok so i changed that now to:
```
assert.Eventually(t, func() bool {
		return app.LogBroadcaster.TrackedAddressesCount() >= 2
	}, 3*time.Second, 200*time.Millisecond)
```
after adding that new method. should be much more clear what we are waiting for